### ISSUE TITLE
WebSocket connection failing

### ISSUE THREAD

**Issue by michaelchadwick** [julienbourdeau/debugbar#14]
I'm trying this awesome project out on a Rails 7.1.3 application, but getting the following error in the dev console:

```
WebSocket connection to 'ws://127.0.0.1:3000/_debugbar/cable' failed: 
```

I think this has something to do with Action Cable, but I'm only moderately familiar with how Ruby on Rails works, and have never used Action Cable before. What might be wrong?

**Comment by julienbourdeau** [comment]
Did you configure `config.action_cable.allowed_request_origins` as shown at the bottom of this page?
https://debugbar.dev/docs/installation/

**Comment by michaelchadwick** [comment]
The code in your installation guide mentions something called Algolia, which I don't use, so I'm not sure what to put in place of it. My site uses Redis, but switching that in did not work.

**Comment by julienbourdeau** [comment]
Yeah the doc is incorrect. It’s because I work at Algolia and I copy
pasted the name from the codebase I’m working on. It should be your app
name. You can probably use Rails:: instead of Algolia::

I’ll edit the docs. 

On February 22, 2024, Marcin Magnus ***@***.***> wrote:
> The code in your installation guide mentions something called Algolia,
> which I don't use, so I'm not sure what to put in place of it. My site
> uses Redis, but switching that in did not work.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/julienbourdeau/debugbar/issues/14#issuecomment-
1960109282>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AALUPBFCUOCDVTDHFXNN5TDYU6LY7AVCNFSM6AAAAABDVHTREWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNRQGEYDSMRYGI>.
You are receiving this because you commented.Message ID:
***@***.***>

**Comment by fusionlove** [comment]
Same issue. Would appreciate fix so can try software.

**Comment by michaelchadwick** [comment]
While testing, I serve it using `rails s`.

**Comment by vkazakevich** [comment]
@michaelchadwick try specifying the url of your local site for callback `debugbar_javascript` ERB helper.

Like this:
`<%= debugbar_javascript cable: {url: "ws://localhost:3000"} %>`

**Comment by sergiopantoja** [comment]
@vkazakevich That worked for me! Thanks!

**Comment by michaelchadwick** [comment]
@vkazakevich That fixed that error! However, now I get this error in the dev console: `Uncaught SyntaxError: Identifier 'cS' has already been declared (at script:1:1)`

**Comment by Eric-Guo** [comment]
Can also using `config.action_cable.disable_request_forgery_protection = true` which maybe easier then `config.action_cable.allowed_request_origins = [%r{http://*}, %r{https://*}]`

**Comment by julienbourdeau** [comment]
@Eric-Guo Yeah it's definitely easier, I updated the docs. Thanks!

@michaelchadwick This error is fixed in 0.3.0, see #26 

@vkazakevich Thank you! I added a troubleshooting section to the docs with your fix 🙏

**Comment by jathayde** [comment]
I had further issues as I went down this debug path. Found that the AppSignal logger dumping setup was messing with ActionCable. I updated my initializer that had other logging stuff in it based on this thread (https://github.com/bensheldon/good_job/issues/596#issuecomment-1123914217):

```
formatter = ActiveSupport::Logger::SimpleFormatter.new
formatter.extend ActiveSupport::TaggedLogging::Formatter
Rails.logger.formatter = formatter
```

And now it's working great. Adding here for anyone else that hits this weird conflict between semantic logger and debugbar
